### PR TITLE
Add eve unit tests and update tool configs

### PR DESCRIPTION
**Pull request by egparedes**
## Description

Migrate missing eve unit tests from gtc repository and update tooling configurations accordingly.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -176,3 +193,24 @@ allow_untyped_defs = False
...
+    lit_suite: pytest item running a LLVM-lit test suite
+norecursedirs =
+    dist
+    build
+    _local*
+    .*
+testpaths = tests
+
+
+#-- sphinx --
```
Comment (havogt) [inline setup.cfg:212]: commented on purpose?

**Review by havogt — COMMENTED** [review]
Only one question.

**Comment by havogt** [comment]
I merged the gitlabci PR. There is now a conflict. Also we should run `bors try` after conflict resolution to be sure changes don't break it.

File: setup.cfg (hunk 1 above)
Comment (egparedes) [reply setup.cfg:212]: Yes, this data is out of sync until we change the repo structure to have a distribution package per subproject.

**Comment by egparedes** [comment]
bors try

**Review by havogt — APPROVED** [review]

**Pull request merged**